associated-token-account/program/src/processor.rs

```diff
@@ -157,3 +160,148 @@ fn process_create_associated_token_account(
         ],
     )
 }
+
+/// Processes `CloseNested` instruction
+pub fn process_close_nested(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
```

Comment:
nit: it looks like this instruction doesn't correctly close the nested account (unlike the create family ixes) if deployed to unofficial address (not `ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL`), according to its own independent address space.  i know it's a bit unusual thing with no conceivable use case.

how about making this `_program_id` and hard-code `id()` in `get_associated_token_address_and_bump_seed_internal()` invocation below?

Alternatively, we can stop calling the handy `get_associated_token_address_with_program_id()` and correctly pass this `program_id` to `get_associated_token_address_and_bump_seed_internal()`. but i wonder the merit of it...

Anyway, current code is a bit mixture of `program_id` and (implicit use of) `id()`, which is a bit confusing. (hehe, i tried to attack with this fact. but no avail)

Reply:
Great catch, I'll fix this up to be consistent with create, and avoid using `id()` anywhere in the program, *just in case* we decide to deploy another ATA program.